Source/DafnyCore/AST/TokenRangeOrigin.cs

```diff
@@ -0,0 +1,57 @@
+#nullable enable
+using System;
+
+namespace Microsoft.Dafny;
+
+[method: SyntaxConstructor]
+public class TokenRangeOrigin(Token startToken, Token endToken) : IOrigin {
```

Comment:
This will certainly be a nice convenience for tools producing serialized Dafny programs, but `TokenRangeOrigin` appears equivalent to the `SourceOrigin(Token, Token)` constructor - do you think it could be simpler to support multiple `SyntaxConstructor`s on an AST type?

Reply:
> do you think it could be simpler to support multiple SyntaxConstructors on an AST type?

I think it would be nicer for the consumer of serialization library (Dafny being the consumer here), but we'd have to figure out what it would look like in the schema. I think in the schema you still want a separate type. The best way I see of enabling it would be to let every `SyntaxConstructors` attribute after the first take a string argument, which would be the name of the extra type in the schema.

Any base classes of a type with multiple `SyntaxConstructors`, would use 'primary' one, the one without an extra argument.